src/styles/mixins/use-logical.scss

```diff
@@ -1,48 +1,120 @@
+    functions.var($element, get-logical-name($property, -block-start), nth($logical-values, 1))
+    functions.var($element, get-logical-name($property, -block-end), nth($logical-values, 3))
+  );
+  #{get-logical-name($property, -inline, $element-to-overwrite)}: functions.var($element, get-logical-name($property, -inline),
+    functions.var($element, get-logical-name($property, -inline-start), nth($logical-values, 4))
+    functions.var($element, get-logical-name($property, -inline-end), nth($logical-values, 2))
+  );
+}
+
+@mixin use-logical($element, $property, $values, $element-to-overwrite: undefined) {
   @if $values and $property {
-    $includeBorderRadius: str-index($property, border-radius);
-    @if $includeBorderRadius {
-      $options: start-start, start-end, end-start, end-end;
-      $logicalValues: getLogicalValues($values);
-      @for $index from 1 through 4 {
-        border-#{nth($options, $index)}-radius: functions.var($element, border-#{nth($options, $index)}-radius, nth($logicalValues, $index));
-      }
+    $include-border-radius: str-index($property, border-radius);
+
+    @if $include-border-radius {
+      @include border-radius($element, $values, $element-to-overwrite);
     } @else {
-      $logicalValues: getLogicalValues($values);
-      #{getLogicalName($property, -block)}: functions.var($element, getLogicalName($property, -block),
-        functions.var($element, getLogicalName($property, -block-start), nth($logicalValues, 1))
-        functions.var($element, getLogicalName($property, -block-end), nth($logicalValues, 3))
-      );
-      #{getLogicalName($property, -inline)}: functions.var($element, getLogicalName($property, -inline),
-        functions.var($element, getLogicalName($property, -inline-start), nth($logicalValues, 4))
-        functions.var($element, getLogicalName($property, -inline-end), nth($logicalValues, 2))
-      );
+        @include default($element, $property, $values, $element-to-overwrite);
     }
   }
 };
+
+@mixin overwrite-border-radius($element, $values) {
+  $options: start-start, start-end, end-start, end-end;
+  $logical-values: get-logical-values($values);
+
+  @for $index from 1 through 4 {
+    --#{$element}-border-#{nth($options, $index)}-radius: #{nth($logical-values, $index)};
+  }
+}
+
+@mixin overwrite-default($element, $property, $values) {
+  $logical-values: get-logical-values($values);
+
+  $block-values: nth($logical-values, 1)!default;
+  @if (nth($logical-values, 3) != "0") {
+    $block-values: #{$block-values} #{nth($logical-values, 3)};
+  }
+
+  $inline-values: nth($logical-values, 4)!default;
+  @if (nth($logical-values, 2) != "0") {
+    $inline-values: #{$inline-values} #{nth($logical-values, 2)};
+  }
+
+  #{get-logical-name($property, -block, $element)}: $block-values;
+  #{get-logical-name($property, -inline, $element)}: $inline-values;
+}
+
+@mixin overwrite-logical($element, $property, $values) {
```

Comment:
IMO overwrite should be used when you want to overwrite CSS var but without new values. IMO it's real overwiten